### ISSUE TITLE
ArrowIOError: Broken pipe

### ISSUE THREAD

**Issue by iamtodor** [nalepae/pandarallel#40]
I have encountered `ArrowIOError: Broken pipe`
I was applying `parallel_apply` to first 50 values within pd.Series

```python
def make_request(uid):
    uid = int(uid)
    url = some_endpoint
    response = requests.post(url).json()
    return response['total']
```
```
---------------------------------------------------------------------------
ArrowIOError                              Traceback (most recent call last)
/usr/local/lib/python3.7/site-packages/pandarallel/utils.py in wrapper(*args, **kwargs)
     74             try:
---> 75                 return func(*args, **kwargs)
     76 

/usr/local/lib/python3.7/site-packages/pandarallel/series.py in closure(series, func, *args, **kwargs)
    148 
--> 149             object_id = plasma_client.put(series)
    150 

/usr/local/lib/python3.7/site-packages/pyarrow/_plasma.pyx in pyarrow._plasma.PlasmaClient.put()

/usr/local/lib/python3.7/site-packages/pyarrow/_plasma.pyx in pyarrow._plasma.PlasmaClient.create()

/usr/local/lib/python3.7/site-packages/pyarrow/error.pxi in pyarrow.lib.check_status()

ArrowIOError: Broken pipe

During handling of the above exception, another exception occurred:

ArrowIOError                              Traceback (most recent call last)
<ipython-input-17-95842d7aa72c> in <module>
      1 start = datetime.now()
      2 print(start)
----> 3 data[1].iloc[:50].parallel_apply(make_request, axis=1)
      4 print('time elapsed', datetime.now() - start)

/usr/local/lib/python3.7/site-packages/pandarallel/utils.py in wrapper(*args, **kwargs)
     84 
     85             finally:
---> 86                 client.delete(client.list().keys())
     87 
     88         return wrapper

/usr/local/lib/python3.7/site-packages/pyarrow/_plasma.pyx in pyarrow._plasma.PlasmaClient.list()

/usr/local/lib/python3.7/site-packages/pyarrow/error.pxi in pyarrow.lib.check_status()

ArrowIOError: Broken pipe
```

**Comment by larsmathuseck** [comment]
Same here, happens in JupyterLab.
Any Idea?

**Comment by nalepae** [comment]
I'm currently developing a new version of **Pandarallel** without **PyArrow Plasma** (which seems to be the cause of your bug).

This version is not yet relased, but you can already try it by:
- Cloning this git repository on your computer
- Switching to **develop** branch
- Running `$ pip install .` (with the dot after the install)

By default, this version of **Pandarallel** will try to use `/dev/shm` to transfer data between the main process and the workers. If you don't want to use this feature but you prefer to use standard multiprocessing tranfert feature (pipe), you can disable it by passing `use_memory_fs_is_available=False` in the initialize method.

See the docstring of `initialize` for more information.

Note the `shm_size_mb` parameter of `initialize` is now deprecated since **Pandarallel** don't use **PyArrow** anymore.

In you chosoe to use `/dev/shm` to transfer data between the main process and the workers and you got a memory error, you can either:
- Increase the size of this partition, thanks to [this link](https://tontr.blogspot.com/2016/07/how-to-resize-tmpfs-devshm-in-linux.html), or
- Use standard multiprocessing tranfert feature (pipe), by passing `use_memory_fs_is_available=False` in the initialize method.

Note that I did **not** re-implemented (yet) progress bar and verbosity option in this version.

To remove this develop version and retrieve the official one:
- `$ pip uninstall pandarallel`
- `$ pip install pandarallel`

Please let me know if you encounter a new bug or if it works better now.

Regards,

Manu

**Comment by nalepae** [comment]
Fixed with `pandarallel 1.4.0`

It seems your bug comes from the usage of `pyarrow plasma`.

`pandarallel 1.4.0` does not use `pyarrow plasma` any more.
